### ISSUE TITLE
insights: show on the creation/edit screen what dashboard (and its permissions level) the insight appears on

### ISSUE THREAD

**Issue by Joelkw** [sourcegraph/sourcegraph#30603]
Valid feedback from users that without this indication in the UI [you're not always sure which dashboard your insight is created on](https://sourcegraph.productboard.com/insights/notes/all-notes/notes/21474253), and that you want [to be sure that it's private if it needs to be](https://sourcegraph.productboard.com/insights/notes/all-notes/notes/21474367). 

We have old original designs but have not implemented yet, and perhaps they need to be worked into the [designs for the "# of dashboards" modal](https://www.figma.com/file/db3qb0UjYLo9RYFxQYjWXz/Show-how-many-dashboards-have-the-current-editing-insight-%2329943-Review?node-id=275%3A1628). cc @AlicjaSuska

**Comment by vovakulikov** [comment]
I told this already somewhere in our designs/gh issues but I also want to drop about it once again here since it looks like the main place for this change. 

I think that instead of just description copy we should make the insight visibility section actionable. Like Combobox UI where you could add or remove any dashboard that is accessible to you (that means include insight to these dashboards). For dashboards that are not accessible to you due to the permission system, we could just render disabled "gray"/"unknown" combo box pills 

<img width="741" alt="Screenshot 2022-02-03 at 13 30 12" src="https://user-images.githubusercontent.com/18492575/152325526-9a80dfe8-bea6-422b-af71-36d3de0ea03b.png">

What problem does this solve? Well, let's imagine that we're browsing insight and you clicked the "Create insight" button and by the current logic by this you're going to create insight for this dashboard but what if you just want to create insight for another dashboard or include this insight it multiples dashboard at once. To do this now you have to create insight on the right dashboard then go to those multiple dashboards and add a newly created insight manually. It's just too many steps for such a simple operation. I think that this Combobox solution solves this problem and also improves readability over our dashboard permission system.

cc @Joelkw @AlicjaSuska

**Comment by Joelkw** [comment]
@vovakulikov I like this proposal – you did a good job noting the need for an ability to adjust (more than one, possibly) dashboards on the creation/edit screen. 

My only question is whether a combobox is the right UI interaction for this, cc @AlicjaSuska . Also, do the tile of the combo box contain the permissions level as well as the title?

**Comment by vovakulikov** [comment]
@Joelkw Why do I believe that Combobox here is a good fit 

- Combobox allows you to modify (delete or add) dashboards (Actually this means adding insight to these dashboards) 
- Also it has a disabled state where you can't see the dashboard title or permissions but see that this insight has been included in some other inaccessible to you dashboards. (It actually doesn't mean that we don't need to have a warning message about editing insight when it is included in such dashboards) 

> Also, do the tile of the combo box contain the permissions level as well as the title

It could be, here I pass it to @AlicjaSuska but we could hide details information in the tooltip if the tile doesn't have enough space there. 

The main concern here is that I remember @coury-clark mentioned that this insights/dashboard connection isn't necessarily an easy thing to do on the backend. But we could start from read-only combobox here and then make it interactive when our BE will be ready for including insights to dashboards through insight-specific GQL mutation. (At the moment the only way to do it is using a dashboard specific API - so it's one way street)

**Comment by Joelkw** [comment]
@AlicjaSuska tagging this for milestone 18 (mid-may iteration) – do you think we could get quick designs (or pull what we had before) so this is generally ready to put into engineering when we have capacity (may be 18, maybe iteration 19-20)?

**Comment by Joelkw** [comment]
cc @AlicjaSuska checking in on this – what are the current state of designs?

**Comment by AlicjaSuska** [comment]
@Joelkw I'll be working on it this week. I agree with you and @vovakulikov that combobox is a good solution there, I need to think over UX the details like dashboards that users cannot access and showing permissions. Will update this issue with a file.

**Comment by Joelkw** [comment]
@AlicjaSuska just checking on this again – what's the best way to get this prioritized? It seems like a small design effort, and it'd be awesome if we can have this ready so we can ship this in Q2, if not in June.

**Comment by AlicjaSuska** [comment]
@Joelkw yes, I'll share it today. It's basically ready, need to polish figma.

**Comment by Joelkw** [comment]
@AlicjaSuska my apologies – where can I find the designs for this?

**Comment by AlicjaSuska** [comment]
@Joelkw here are all the materials ready for review:
- [Figma](https://www.figma.com/file/qrxCfN2EkkEH5YyJm9b31V/?node-id=1625%3A8522)
- [Loom](https://www.loom.com/share/0c79dbc6545e45f2a8a5b8764b100856)

**Comment by AlicjaSuska** [comment]
This issue is ready for dev - I've updated the description.

**Comment by felixfbecker** [comment]
@Joelkw are we prioritizing this issue for this iteration?

**Comment by Joelkw** [comment]
No, let's focus on the compute and unification improvements for now given my updated understanding of our capacity. Thanks.

**Comment by vovakulikov** [comment]
@sourcegraph/code-insights-backend I have a really big consers about fetching all dashboards for the creation and edit UI. We should extend our API in order to support search/filter by string 

Relevant Figma thread https://www.figma.com/file/qrxCfN2EkkEH5YyJm9b31V?node-id=1694:14923#207741640

**Comment by vovakulikov** [comment]
From sync with @chwarwick 
- it seems that supporting filter field for the dashboard API should be 3-5 in estimation points

Possible API (we should extend insightsDashboards resolver with search and excludeIDs arguments) 
```
 insightsDashboards(first: Int, after: String, id: ID, search: String!, excludeIDs: [ID!]): InsightsDashboardConnection!
```

**Comment by coury-clark** [comment]
@vovakulikov could you elaborate what data you would imagine `search` and `excludeIDs` to contain?

**Comment by vovakulikov** [comment]
@coury-clark search is string value from the dashboard combobox field in order to filter dashboard and provide a suggestions list based on input value. Exclude ids is a list of dahsobard ids that were already picked in order to avoid duplicates in the suggestions.

**Comment by christinelovett** [comment]
[Customer](https://github.com/sourcegraph/accounts/issues/1503) had the following feedback:

> It's impossible to know the visibility of an insight because that field is not shown on the insight at all

**Comment by chwarwick** [comment]
Just noting again since this came up last time we looked at this.  It's possible to not know the full visibility of the insight because you may not have access to all the dashboards it has been added to.

**Comment by coury-clark** [comment]
backlog

**Comment by coury-clark** [comment]
if there is no backend work we'd like p1

**Comment by coury-clark** [comment]
sounds like backend work is we need a dashboard search respecting user permissions
